src/app/main/component/user/components/profile/user-notifications/user-notifications.component.ts

```diff
@@ -260,47 +263,81 @@ export class UserNotificationsComponent implements OnInit, OnDestroy {
     }
   }
 
-  acceptRequest(userId: number): void {
-    let isAccepted = true;
-    this.userFriendsService.acceptRequest(userId).subscribe({
-      error: () => {
-        isAccepted = false;
-      },
-      complete: () => {
-        if (isAccepted) {
-          this.matSnackBar.openSnackBar('friendInValidRequest');
+  acceptRequest(notification: NotificationModel): void {
+    if (notification.notificationType === this.notificationFriendRequest) {
+      let isAccepted = true;
+
+      const userId = notification.actionUserId[0];
+
+      this.userFriendsService.acceptRequest(userId).subscribe({
+        error: () => {
+          isAccepted = false;
+        },
+        complete: () => {
+          if (isAccepted) {
+            this.matSnackBar.openSnackBar('friendInValidRequest');
+          }
```

Comment:
There is no feedback in case of failure/else, which can lead to confusion, but need look to the UserStory